blockindex/bloomfilterindexer.go

```diff
@@ -216,28 +233,82 @@ func (bfx *bloomfilterIndexer) FilterBlocksInRange(l *filter.LogFilter, start, e
 	if endIndex, err = bfx.getIndexByHeight(end); err != nil {
 		return nil, err
 	}
-	br, err := newBloomRange(bfx.bfSize, bfx.bfNumHash)
-	if err != nil {
+
+	var (
+		ctx, cancel = context.WithTimeout(context.Background(), _queryTimeout)
+		blkNums     = make([][]uint64, endIndex-startIndex+1)
+		jobs        = make(chan jobDesc, endIndex-startIndex+1)
+		eg          *errgroup.Group
+		bufPool     sync.Pool
+	)
+	defer cancel()
+	eg, ctx = errgroup.WithContext(ctx)
+
+	// create pool for BloomRange object reusing
+	if _, err := newBloomRange(bfx.bfSize, bfx.bfNumHash); err != nil {
 		return nil, err
 	}
-	for ; startIndex <= endIndex; startIndex++ {
-		bfKey := byteutil.Uint64ToBytesBigEndian(startIndex)
-		if err := bfx.loadBloomRangeFromDB(br, bfKey); err != nil {
-			return nil, err
-		}
-		if l.ExistInBloomFilterv2(br.BloomFilter) {
-			searchStart := br.Start()
-			if start > searchStart {
-				searchStart = start
+	bufPool = sync.Pool{
+		New: func() interface{} {
+			br, _ := newBloomRange(bfx.bfSize, bfx.bfNumHash)
+			return br
+		},
+	}
+
+	for w := 0; w < _workerNumbers; w++ {
+		eg.Go(func() error {
+			for {
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				default:
+					job, ok := <-jobs
+					if !ok {
+						return nil
+					}
+
+					br := bufPool.Get().(*bloomRange)
```

Comment:
there will only be one instance of `*bloomRange`, which will be an issue to be shared by `_workerNumbers` jobs.

Reply:
add `bufPool.Put(br)` on L272